framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -838,6 +854,22 @@ def set_monitor_params(self, current_monitor, **data):
                 SimulatorWizzardURLs.SET_MONITOR_PARAMS_URL, current_monitor)
             return rendering_rules.to_dict()
 
+        if isinstance(monitor, Projection) and cherrypy.request.method == 'POST':
```

Comment:
@popaula937 Maybe you know what happened here at rebase. Is this still necessary?

Reply:
This should not be necessary since at this point we already have the correct GIDs as strings in `data`. And the `fill_from_post` is now handling the conversion of GIDs from string to UUID for the `TraitDataTypeSelectField`.

Reply:
@VinczeRobert can you detail a bit on what is not working?

Reply:
In fact it's not working when you don't choose a RM, so I discussed with Paula to make that field required and in case a surface is chosen, I will add a filter to only allow the region mapping which was selected at that step.

Reply:
Discussed with Robert and the error he was: 
`Please specify a region mapping on the EEG/MEG/iEEG monitor when performing a region simulation.
`
This happens because the RegionMapping was optional on the Projection Monitor, but it is actually required to select one for region-level simulations. Thus, if he did not select a RegionMapping before launching the simulation, he got this error.
It works now because he declared the RegionMapping as required on the Projection monitor, thus, the select field is filled in by default (not because of putting back this code).
Not sure whether we should keep it required, but if we would, we could just ensure that for Surface simulations, the same RM is chosen as for the Cortex.
@liadomide what do you think?

Reply:
Exactly. After reading the code in the library I think we should put this RM required only in the web GUI, as otherwise the library api will change in a not so efficient manner (the user will need to duplicately put RM both on the simulator.cortex and on simulator.monitors[Projection].rm

Reply:
I saw the required flag on RM reverted in the library. Is it put TRUE in the ViewModel  or another place from the framework?

Reply:
It is FALSE now in ProjectionViewModel. I understood that I have to keep it False because it is not required in case of Surface Simulations. 

My past idea was to keep it required and if we have a surface simulation then there would be a filter to only allow the RM already chosen, but I think it wasn't accepted.

Reply:
It was accepted with the observation that we should not change the flag directly in tvb-library, but only in ProjectionViewModel 
> put this RM required only in the web GUI (from Lia)

And yes, show only the already chosen RM, and maybe disable the field entirely.